From 31d7ddf7b0aee8e55dd17fb56dbda87f9108fd4a Mon Sep 17 00:00:00 2001
From: Rahul Anand <ranand@codeaurora.org>
Date: Mon, 12 May 2014 13:43:59 -0700
Subject: [PATCH] QR-Linux Patches Mods to get this working on QR-Linux New
 change: modifed start_usb and start_adbd to not start on
 any runlevel, since there's an upstart job that calls them

Change-Id: Idcbe0d146d7f0e01be603b9a7e76b61007b3c6ae
---
 adb/Makefile.am       |   38 +++-----------------------------------
 adb/start_adbd        |   14 +++++++++++++-
 configure.ac          |    2 +-
 libcutils/Makefile.am |   41 +++++++++++++++++++++++++++++++++++++++++
 usb/Makefile.am       |   20 ++++++++++----------
 usb/compositions/9025 |   16 +++++++++++++++-
 usb/start_usb         |   15 ++++++++++++++-
 usb/target            |    3 +++
 8 files changed, 100 insertions(+), 49 deletions(-)

diff --git a/adb/Makefile.am b/adb/Makefile.am
index 5dc9966..8d94dc1 100755
--- a/adb/Makefile.am
+++ b/adb/Makefile.am
@@ -1,39 +1,7 @@
-bin_PROGRAMS = adbd adb
+initddir = /etc/init.d
+initd_SCRIPTS = start_adbd
 
-# adb host tool for linux
-# =========================================================
-adb_SOURCES := \
-	adb.c \
-	console.c \
-	transport.c \
-	transport_local.c \
-	transport_usb.c \
-	commandline.c \
-	adb_client.c \
-	sockets.c \
-	services.c \
-	file_sync_client.c  \
-	utils.c \
-	usb_vendors.c \
-	fdevent.c \
-	usb_linux.c \
-	get_my_path_linux.c
-
-adb_CFLAGS = -O2 \
-	-g \
-	-DADB_HOST=1 \
-	-Wall \
-	-Wno-unused-parameter \
-	-D_XOPEN_SOURCE \
-	-D_GNU_SOURCE \
-	-I../include \
-	-include ../include/arch/linux-arm/OEConfig.h
-
-adb_LDFLAGS = -lpthread -all-static
-
-adb_LDADD = ../libcutils/libcutils.a
-adb_LDADD += ../libzipfile/libzipfile.a
-adb_LDADD += -lz
+bin_PROGRAMS = adbd
 
 # adbd device daemon
 # =========================================================
diff --git a/adb/start_adbd b/adb/start_adbd
index c4b1feb..1a3d2a1 100644
--- a/adb/start_adbd
+++ b/adb/start_adbd
@@ -1,4 +1,16 @@
 #! /bin/sh
+### BEGIN INIT INFO
+# Provides:          start_adbd
+# Required-Start:    
+# Required-Stop:     
+# Should-Start:
+# Should-Stop:
+# Default-Start:     
+# Default-Stop:      
+# Short-Description: Start android debug bridge deamon
+# Description:       This script enabls android debug bridge over usb.
+#                    Does not start/stop on any runlevel. Called from upstart job only
+### END INIT INFO                                                       
 #
 # Copyright (c) 2011, The Linux Foundation. All rights reserved.
 #
@@ -38,7 +50,7 @@ fi
 case "$1" in
   start)
         echo -n "Starting adbd: "
-        start-stop-daemon -S -b -a /sbin/adbd
+        start-stop-daemon -S --exec /usr/bin/adbd -b start_adbd
 
         if [ -e "/etc/adb_devid" ]
         then
diff --git a/configure.ac b/configure.ac
index 1886c58..7a9e0fe 100755
--- a/configure.ac
+++ b/configure.ac
@@ -53,7 +53,7 @@
           [HOST_OS= 'linux'])
 
 
-  if test "x$HOST_OS" = "xlinux-gnueabi"; then
+  if test "x$HOST_OS" = "xlinux-gnueabihf"; then
 	HOST_OS=linux
   else
 	AC_MSG_NOTICE([Unknown host OS $HOST_OS detected.])
diff --git a/libcutils/Makefile.am b/libcutils/Makefile.am
index 81c062f..bd6075b 100644
--- a/libcutils/Makefile.am
+++ b/libcutils/Makefile.am
@@ -1,3 +1,44 @@
+androidincludedir = $(includedir)/android
+
+androidinclude_HEADERS = $(abs_top_srcdir)/include/android/log.h
+
+cutilsincludedir = $(includedir)/cutils
+
+cutilsinclude_HEADERS = $(abs_top_srcdir)/include/cutils/abort_socket.h \
+	$(abs_top_srcdir)/include/cutils/array.h \
+	$(abs_top_srcdir)/include/cutils/ashmem.h \
+	$(abs_top_srcdir)/include/cutils/atomic-arm.h \
+	$(abs_top_srcdir)/include/cutils/atomic.h \
+	$(abs_top_srcdir)/include/cutils/atomic-inline.h \
+	$(abs_top_srcdir)/include/cutils/atomic-x86.h \
+	$(abs_top_srcdir)/include/cutils/compiler.h \
+	$(abs_top_srcdir)/include/cutils/config_utils.h \
+	$(abs_top_srcdir)/include/cutils/cpu_info.h \
+	$(abs_top_srcdir)/include/cutils/dir_hash.h \
+	$(abs_top_srcdir)/include/cutils/event_tag_map.h \
+	$(abs_top_srcdir)/include/cutils/hashmap.h \
+	$(abs_top_srcdir)/include/cutils/iosched_policy.h \
+	$(abs_top_srcdir)/include/cutils/jstring.h \
+	$(abs_top_srcdir)/include/cutils/logd.h \
+	$(abs_top_srcdir)/include/cutils/logger.h \
+	$(abs_top_srcdir)/include/cutils/log.h \
+	$(abs_top_srcdir)/include/cutils/logprint.h \
+	$(abs_top_srcdir)/include/cutils/memory.h \
+	$(abs_top_srcdir)/include/cutils/misc.h \
+	$(abs_top_srcdir)/include/cutils/mq.h \
+	$(abs_top_srcdir)/include/cutils/mspace.h \
+	$(abs_top_srcdir)/include/cutils/native_handle.h \
+	$(abs_top_srcdir)/include/cutils/open_memstream.h \
+	$(abs_top_srcdir)/include/cutils/process_name.h \
+	$(abs_top_srcdir)/include/cutils/properties.h \
+	$(abs_top_srcdir)/include/cutils/record_stream.h \
+	$(abs_top_srcdir)/include/cutils/sched_policy.h \
+	$(abs_top_srcdir)/include/cutils/selector.h \
+	$(abs_top_srcdir)/include/cutils/sockets.h \
+	$(abs_top_srcdir)/include/cutils/threads.h \
+	$(abs_top_srcdir)/include/cutils/tztime.h \
+	$(abs_top_srcdir)/include/cutils/uio.h \
+	$(abs_top_srcdir)/include/cutils/zygote.h
 
 # NOTE: Only supports builds of shared/static libraries for the target
 
diff --git a/usb/Makefile.am b/usb/Makefile.am
index e438691..30b3d8e 100755
--- a/usb/Makefile.am
+++ b/usb/Makefile.am
@@ -1,12 +1,12 @@
-bin_PROGRAMS = usb_composition_switch
+usbdir = /usr/bin/usb
+compdir = /usr/bin/usb/compositions
+initddir = /etc/init.d
 
-# Program to switch USB compositions
-# =========================================================
-usb_composition_switch_CC = @CC@
+usb_SCRIPTS = target
+comp_SCRIPTS = compositions/9025 compositions/empty compositions/hsic_next \
+	compositions/hsusb_next
+initd_SCRIPTS = start_usb
 
-usb_composition_switch_SOURCES = usb_composition_switch.c
-
-usb_composition_switch_CFLAGS = -O2 \
-				-g \
-				-Wall \
-				-Wno-unused-parameter
+install-data-hook:
+	test -L $(DESTDIR)$(usbdir)/boot_hsusb_composition || $(LN_S) $(compdir)/9025 $(DESTDIR)$(usbdir)/boot_hsusb_composition
+	test -L $(DESTDIR)$(usbdir)/boot_hsic_composition || $(LN_S) $(compdir)/empty $(DESTDIR)$(usbdir)/boot_hsic_composition
diff --git a/usb/compositions/9025 b/usb/compositions/9025
index af2ebea..fbc47e1 100755
--- a/usb/compositions/9025
+++ b/usb/compositions/9025
@@ -34,6 +34,17 @@ else
 	num="0"
 fi
 
+run_8960 () {
+    echo 0 > /sys/class/android_usb/android$num/enable
+    echo 0x9025 > /sys/class/android_usb/android$num/idProduct
+    echo 0x05C6 > /sys/class/android_usb/android$num/idVendor
+    echo diag > /sys/class/android_usb/android0/f_diag/clients
+    echo smd,tty > /sys/class/android_usb/android0/f_serial/transports
+    echo smd,bam > /sys/class/android_usb/android0/f_rmnet/transports
+    echo diag,adb,serial,rmnet,mass_storage > /sys/class/android_usb/android$num/functions
+    echo 1 > /sys/class/android_usb/android$num/enable
+}
+
 run_9x15() {
 	echo 0 > /sys/class/android_usb/android$num/enable
 	echo 0x9025 > /sys/class/android_usb/android$num/idProduct
@@ -67,7 +78,10 @@ run_9x25_v2() {
 	echo 1 > /sys/class/android_usb/android$num/enable
 }
 
-case `source /usr/bin/usb/target` in
+case `/usr/bin/usb/target` in
+        *8960* )
+                run_8960 &
+                ;;
 	*9x15* )
 		run_9x15 &
 		;;
diff --git a/usb/start_usb b/usb/start_usb
index 2e3ce5e..5d81183 100755
--- a/usb/start_usb
+++ b/usb/start_usb
@@ -1,4 +1,17 @@
 #!/bin/sh
+### BEGIN INIT INFO
+# Provides:          start_usb
+# Required-Start:
+# Required-Stop:
+# Should-Start:
+# Should-Stop:
+# Default-Start:     
+# Default-Stop:      
+# Short-Description: Configure usb composition for android debug bridge
+# Description:       This script configures usb composition to 9025 mode
+#                    i.e. diag,serial_smd,serial_tty,rmnet_bam,mass_storage,adb.
+#                    Does not start/stop on any runlevel. Called from upstart job only
+### END INIT INFO                                                       
 #
 # Copyright (c) 2012-2013, The Linux Foundation. All rights reserved.
 #
@@ -33,7 +46,7 @@ case "$1" in
 
         case $KERNEL in
           3.*)
-		case `source /usr/bin/usb/target` in
+		case `/usr/bin/usb/target` in
 			*9x15* )
 			# Nothing to do here for 9x15
 			;;
diff --git a/usb/target b/usb/target
index a6cb631..367383a 100644
--- a/usb/target
+++ b/usb/target
@@ -26,6 +26,9 @@
 
 
 case `cat /sys/devices/system/soc/soc0/build_id` in
+        *8960* )
+                res="8960"
+                ;;
 	*9x15* )
 		res="9x15"
 		;;
-- 
1.7.9.5

